pywbem_mock/_wbemconnection_mock.py

```diff
@@ -356,14 +378,91 @@ def __repr__(self):
     #
     ################################################################
 
+    def add_namespace(self, namespace):
+        """
+        Add a CIM namespace to the mock repository.
+
+        The namespace must not yet exist in the mock repository.
+
+        Note that the default connection namespace automatically exists in the
```

Comment:
This comment may require some more explanation.  I had to think about it for a moment.  Maybe something like "It is created when the _wbemconnection_mock is constructed."  Also might better refernce with conn.default_namespace.

Question. What happens if you do delete_namespace on the default conneciton namespace

Reply:
On  the comment: Changed that to: "Note that the default connection namespace is automatically added to the mock repository upon creation of this object."

On deleting the default connection namespace: There is no externally available `delete_namespace()` method on the `FakedWBEMConnection` class (only an internal `_remove_namespace()`). However, the faked DeleteInstance operation would remove it. I have added a check in the internal `_remove_namespace()` method that raises CIMError(CIM_ERR_NAMESPACE_NOT_EMPTY) if the default connection namespace is attempted to be removed, and a description of that behavior in the faked DeleteInstance operation.